### ISSUE TITLE
Unify the names and establish a guideline for contributors

### ISSUE THREAD

**Issue by YuanfengZhang** [pola-rs/polars#8862]
### Problem description

Here is just a tiny improvement you may consider.

Some new and old names of parameters and methods are inconsistent. We know that the old ones were made to reduce the difficulty of migration from pandas to polars for users, and the new ones are more clear, direct and easier to understand. But as the codebase grows, the inconsistency will disturb the contributors and users more and more. For example, the "sheet_id", "sheet_name" of read_excel() (old, same as pandas) and the "worksheet" of DataFrame.write_excel() (new, more clear and direct) are inconsistent.

Not only the names among different functions in API but also the name of same parameter suffers from alike inconsistency. For example, the "sep" (old, same as pandas) and the "separator" (new, more clear and direct) of some I/O functions in API. Changing it once maybe okay, but changing without "permanent decision" tag is troublesome.

There are two todos can help this:
1. Review the API reference → list the conflicts → discuss the principles and choices → solve all the conflicts once.
2. Summarize the principles and details as a guideline for contributors, make sure that the names from different contributors are identical (no conflicts) and permanent (no need to change anymore).

Wish polars continues to thrive and prosper!~~~

**Comment by deanm0000** [comment]
>  solve all the conflicts once.

an idealist, I see.

**Comment by YuanfengZhang** [comment]
> > solve all the conflicts once.
> 
> an idealist, I see.

No, sir. Actually it's feasible. I mean the #7533 and alike issues should be solved and merged in one thread instead of proposed and applied from time to time.

**Comment by alexander-beedie** [comment]
I think you'll find no objection to unifying anything that hasn't been addressed already - if you know where some inconsistencies are, let's look at fixing them  ;)

**Comment by stinodego** [comment]
You can find some guidance [here](https://github.com/pola-rs/polars/issues/6120#issuecomment-1374911565). We should pin that somewhere.

Feel free to suggest specific changes you would like to see. We regularly review the public API to see if we can improve naming of functions or parameters. If there is something we have overlooked, please point it out. Generic advice "make it better" is not all that helpful - believe me, we try!
